prow/cmd/autobump/bump.sh

```diff
@@ -198,17 +198,18 @@ list() {
   fi
 }
 
-split_on_commas() {
-  local IFS=,
-  local array=("$1")
-  echo "${array[@]}"
-}
-
-add_suffix() {
-  local array=("$1")
-  local suffix="${2:-/*.yaml}"
-  echo "${array[@]/%/$suffix}"
-}
+ split_on_commas() {
+   local IFS=,
+   local array
+   read -ra array <<< "$1"
+   echo "${array[@]}"
+ }
+
+ add_suffix() {
+   local array=($1)
```

Comment:
passing arrays to functions is a huge red flag that you shouldn't be using shell anymore.

Reply:
Agree that we should use a better language that can be tested. While in this case it's not really passing arrays to function, it passes a string with space in it, so technically it's still a single parameter

Reply:
if we're relying on word splitting, it's an array :-)

Reply:
in this case it appears to be file paths, so this logic is broken for any path with a space in it.

Reply:
FWIW I intended for this to be written in go originally, but we decided that was taking too long and revamping the bash script would be faster. Here is my proposal/design doc: https://docs.google.com/document/d/1XgmZ7lYLoYWIchpnHYw4rjfImhozpcOxqRJgGrh0sS0/edit?usp=sharing

Reply:
@BenTheElder , I have updated the script solving the problem of space in array. Ran a local test and it worked like this:

```
main() {
  COMPONENT_FILE_DIR="a a,b b,c c"
  local IFS=,
  local component_file_dir_array
  read -ra component_file_dir_array <<< "${COMPONENT_FILE_DIR}"
  bumpfiles=("${component_file_dir_array[@]/%//*.yaml}")
  for f in "${bumpfiles[@]}"; do
    echo "File is: $f"
  done
}

main
```

Output:
```
File is: a a/*.yaml
File is: b b/*.yaml
File is: c c/*.yaml
```